Show messages for draft work versions

nicktiberi opened the pull request:
Fixes #1081 
Fixes #1079 

If a user is viewing a draft work version and there is a published version available, show a message directing the user to the latest published version.

This is an inversion of the previous messaging; we no longer show a message on published versions that a draft version is available.

## Viewing the draft

![Screen Shot 2021-06-24 at 11 14 09 AM](https://user-images.githubusercontent.com/312085/123288255-61e21700-d4dd-11eb-8aac-a84f9492001c.png)

## Viewing an older version

![Screen Shot 2021-06-24 at 11 14 17 AM](https://user-images.githubusercontent.com/312085/123288281-69092500-d4dd-11eb-9893-cd0b476b00a7.png)

mtribone commented:
One creator no remove link

<img width="860" alt="screen shot 2017-11-16 at 3 48 39 pm" src="https://user-images.githubusercontent.com/4163828/32915072-b87711c0-cae5-11e7-8b8a-8dd49949491a.png">

mtribone commented:
Two creators with remove links

<img width="869" alt="screen shot 2017-11-16 at 3 50 24 pm" src="https://user-images.githubusercontent.com/4163828/32915162-f32fab9c-cae5-11e7-8b58-355a49adefb6.png">

mtribone commented:
Existing work with two creators and remove links

<img width="852" alt="screen shot 2017-11-16 at 3 52 08 pm" src="https://user-images.githubusercontent.com/4163828/32915236-318838aa-cae6-11e7-93c9-309b94abc540.png">

mtribone commented:
Removing first creator on an existing work with 2 creators and no remove link


<img width="861" alt="screen shot 2017-11-16 at 3 53 14 pm" src="https://user-images.githubusercontent.com/4163828/32915303-6ad5e896-cae6-11e7-857c-e85d6f1af1a3.png">

cam156 commented on app/views/records/edit_fields/_creator.html.erb:
```diff
@@ -9,14 +9,20 @@
+  <% if @form.creators.count == 1
+       remove_button_style = 'display: none'
+     else
+
```
@mtribone If we are going to be explicit this should be remove_button_style = ''

mtribone replied:
@cam156 ah. copy/pasta. double quotes

cam156 commented on app/views/records/edit_fields/_creator.html.erb:
```diff
@@ -9,14 +9,20 @@
-        <span class="controls-remove-text">Remove</span>
-      </button>
+        <button type="button" class="btn btn-link remove remove-creator" style="<%= remove_button_style %>" >
+          <span class="glyphicon glyphicon-remove"></span><span class="controls-remove-text">Remove</span>
```
@mtribone was there a reason to change the spacing here?  It muddies the PR.

cam156 commented on app/views/records/edit_fields/_creator_template.html.erb:
```diff
@@ -3,7 +3,7 @@
       <span class="glyphicon glyphicon-remove"></span><span class="controls-remove-text">Remove</span>
     </button>
-    <input type="hidden" name="<%= @form.model_class_name %>[creators][{{index}}][id]" id="<%= @form.model_class_name %>{{index}}_id" value="{{id}}">
+      <input type="hidden" name="<%= @form.model_class_name %>[creators][{{index}}][id]" id="<%= @form.model_class_name %>{{index}}_id" value="{{id}}">
```
@mtribone any reason to change this line.  The only thing I can see is spacing and it is not aligned with the code above

nicktiberi committed:
Show messages for draft work versions

If a user is viewing a draft work version and there is a published
version available, show a message directing the user to the latest
published version.

This is an inversion of the previous messaging; we no longer show
a message on published versions that a draft version is available.

nicktiberi commented:
@awead - this is working, but has some failing tests. If you are doing any work tonight, feel free to take this over. Otherwise, I will look into it more tomorrow.

nicktiberi commented on app/components/work_version_detail_component.rb:
```diff
@@ -7,43 +7,23 @@ def initialize(work_version:)
+
   def render?
-    !current_draft_version? && i18n_key.present?
+    work_version.draft? || !current_representative_version?
```
I think this needs to be

`work_version.draft? && !current_representative_version?`

As it stands, if the current version is a draft and the representative version (i.e. we only have 1 version and it's a draft), the message will show.

awead replied:
But we do want that, however, it shouldn't link to the current version because it hasn't been published yet.

awead replied:
![Screen Shot 2021-06-24 at 11 19 11 AM](https://user-images.githubusercontent.com/312085/123289034-17ad6580-d4de-11eb-82b2-2f9772d3f3ee.png)
